lisa/sut_orchestrator/azure/azure.py

```diff
@@ -672,20 +689,39 @@ def _validate_template(
             if result:
                 raise LisaException(f"deploy failed: {result}")
         except Exception as identifier:
+            error_message: str = str(identifier)
             if validate_operation:
+                # validate_operation returned, it means deployments created
+                # successfuly. so check errors from deployments by name.
                 deployment = deployments.get(resource_group_name, AZURE_DEPLOYMENT_NAME)
                 # log more details for troubleshooting
                 if deployment.properties.provisioning_state == "Failed":
-                    errors = deployment.properties.error.details
-                    for error in errors:
-                        log.error(f"failed: {error.code}, {error.message}")
-            raise identifier
+                    details = deployment.properties.error.details
```

Comment:
Is `details` here `None` or an empty list/sequence?  Might be a good idea to consider making it an empty list if it's `None` since this will clean up the code -- this pattern seems to be used throughout the code base.

For instance, it's cleaner to do:

```python
for iter in sequence.that.has.been.obtained.from.a.deeply.nested.structure:
   do_something(iter)
```

Rather than:

```python
sequence = sequence.that.has.been.obtained.from.a.deeply.nested.structure
if sequence:
    for iter in sequence:
        do_something(iter)
```

In this particular case, however, it might be slightly more efficient if, instead of `error_message` being of a string type, it being of a `List[String]` type and be named something else.  The reason is that the `join()` method is called over and over for every detail.  This could be rewritten as something like the following:

```python
error_contexts: List[str]

if validate_operation:
    deployment = ...
    if deployment...state == "Failed":
        error_contexts.extend([
                  f"failed: {detail.code}, {detail.message}" for detail in deployment.properties.error.details
        ])
    elif isinstance(...) and identifier.error and identifier.error.details:
        error_contexts.extend([
                  ...
        ])

raise LisaException('\n'.join(error_contexts))
```

(Or something like this. This doesn't seem to be a hotspot and the inefficiency isn't that much of a problem here.)

Reply:
Thank you, I will update second one. For first one, as we use mypy to limit line to 88. Any way to format long nested names?